Problem6/src/problem6/Problem6.java

```diff
@@ -10,7 +14,35 @@
      * @param args the command line arguments
      */
     public static void main(String[] args) {
-        // TODO code application logic here
+        Scanner scanner = new Scanner(System.in);
+        
+        String input = scanner.nextLine();
+        
+        StringTokenizer tokenizer = new StringTokenizer(input);
+        
+        String finalOutput = "";
+        String currentLine = "";
+        
+        while (tokenizer.hasMoreTokens()) {
+            
+            String token = tokenizer.nextToken();
+            
+            //validate input
+            if (token.length() > 16) {
+                System.out.println("The input includes " + token + ", which is longer than 16 characters.");
+                return;
+            }
+            
+            
+            if ((currentLine + " " + token).length() > 16) {
+                finalOutput += "\n" + currentLine;
+                currentLine = token;
+            } else {
+                currentLine += " " + token;
+            }
+        }
+        
+        System.out.println(finalOutput);
```

Comment:
The final line does not get output.  Using the sample from the question, I get:
```
There are a dozen fuzzy bunnies sleeping under the couch.

 There are a
dozen fuzzy
bunnies sleeping

```


There is also that extra space at the start of the output.